types/ads-client/index.d.ts

```diff
@@ -5,26 +5,26 @@ import EventEmitter = require("events");
 export interface Settings {
     targetAmsNetId: string;
     targetAdsPort: number;
-    objectifyEnumerations?: boolean;
-    convertDatesToJavascript?: boolean;
-    readAndCacheSymbols?: boolean;
-    readAndCacheDataTypes?: boolean;
-    disableSymbolVersionMonitoring?: boolean;
-    routerTcpPort?: number;
-    routerAddress?: string;
-    localAddress?: string;
-    localTcpPort?: number;
-    localAmsNetId?: string;
-    localAdsPort?: number;
-    timeoutDelay?: number;
-    hideConsoleWarnings?: boolean;
-    autoReconnect?: boolean;
-    reconnectInterval?: number;
-    checkStateInterval?: number;
-    connectionDownDelay?: number;
-    allowHalfOpen?: boolean;
-    disableBigInt?: boolean;
-    bareClient?: boolean;
+    objectifyEnumerations?: boolean | null;
+    convertDatesToJavascript?: boolean | null;
+    readAndCacheSymbols?: boolean | null;
+    readAndCacheDataTypes?: boolean | null;
+    disableSymbolVersionMonitoring?: boolean | null;
```

Comment:
General note, but I suspect these should have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

Reply:
@jakebailey thanks, updated in 894e7fc52408e88de754af374e0ab94a874c5fa7